### PR TITLE
OCS endpoint to list remote shares

### PR DESCRIPTION
**Pull request by rullzer**
Required for #9098
- Move OCS endpoint `../remote_shares` -> `../pending_shares`
- Introduce new endpoint `../remote_shares` that allows listing of incoming remote shares.

@nickvergessen since you wrote the original `../remote_shares` stuff are you fine with this?

I'm not really happy with this. But as stated in https://github.com/owncloud/core/issues/9098#issuecomment-143311672 this is the cleanest hack I can think of. Comments?

CC: @cmonteroluque @DeepDiver1975

### CODE REVIEW

File: apps/files_sharing/api/remote.php [hunk 1]
```diff
@@ -90,4 +90,79 @@ public static function declineShare($params) {
...
+			Filesystem::getLoader(),
+			\OC::$server->getHTTPHelper(),
+			\OC::$server->getNotificationManager(),
+			\OC_User::getUser()
+		);
+
+		$shareInfo = $externalManager->getShare($params['id']);
+
+		if ($shareInfo === false) {
+			return new \OC_OCS_Result(null, 404, 'share does not exists');
```
Comment (phil-davis) [inline apps/files_sharing/api/remote.php]: s/exists/exist/

File: apps/files_sharing/api/remote.php [hunk 2]
```diff
@@ -90,4 +90,79 @@ public static function declineShare($params) {
...
+			Filesystem::getLoader(),
+			\OC::$server->getHTTPHelper(),
+			\OC::$server->getNotificationManager(),
+			\OC_User::getUser()
+		);
+
+		$shareInfo = $externalManager->getShare($params['id']);
+
+		if ($shareInfo === false) {
+			return new \OC_OCS_Result(null, 404, 'Share does not exists');
```
Comment (phil-davis) [inline apps/files_sharing/api/remote.php]: s/exists/exist/

**Comment by nickvergessen** [comment]
Maybe remote_shares/pending or do want to use the same endpoint for normal shares in the future?

This needs a PR to fix the URLs in the Pyoc client

**Comment by rullzer** [comment]
> Maybe remote_shares/pending or do want to use the same endpoint for normal shares in the future?

Possible for sharing 2.0 yes.

**Comment by DeepDiver1975** [comment]
Hmmm ... do we really want to change the behavior of already existing public routes?

This is a no go from my perspective - but maybe I missed something.

**Comment by rullzer** [comment]
> 

Hmmm ... do we really want to change the behavior of already existing public routes?
>
> This is a no go from my perspective - but maybe I missed something.

This route was introduced in 8.2. So it is not yet officially released.

**Comment by DeepDiver1975** [comment]
THX. Fine then

File: apps/files_sharing/lib/external/manager.php [hunk 3]
```diff
@@ -180,9 +180,9 @@ private function setupMounts() {
 	 * @param int $id share id
 	 * @return mixed share of false
 	 */
-	private function getShare($id) {
+	public function getShare($id) {
 		$getShare = $this->connection->prepare('
-			SELECT `remote`, `remote_id`, `share_token`, `name`
+			SELECT *
```
Comment (nickvergessen) [inline apps/files_sharing/lib/external/manager.php]: do we want to expose all of the data?
Thinking about password and share_token, I'm not sure they should be leaked like that?

Comment (rullzer) [inline apps/files_sharing/lib/external/manager.php]: So I agree we should limit what we show. But we also show * in the listing of pending shares.. https://github.com/owncloud/core/blob/master/apps/files_sharing/lib/external/manager.php#L400-L430

So this doesn't introduce anything new.

Comment (nickvergessen) [inline apps/files_sharing/lib/external/manager.php]: okay fine by me then

**Comment by MorrisJobke** [comment]
@rullzer Ready to review or not?

**Comment by rullzer** [comment]
@MorrisJobke not yet. I think I need to add some stuff. For #19501 to work properly.

**Comment by nickvergessen** [comment]
btw while preparing a patch for the PyoC Client to change the URL, I realised we can't reuse the same endpoint for remote and normal shares! :bomb: 
The problem is that they have a different unique id. For remote shares we need to accept `oc_share_remote.id` for normal shares `oc_share.id` So they can not reuse the same endpoint.

**Comment by rullzer** [comment]
@nickvergessen warg... unique id's are needed then indeed...

**Comment by rullzer** [comment]
@nickvergessen I guess this is then the most logical solution.

**Comment by rullzer** [comment]
Ok extra info added.. do now you actually know something about the remote shares.

Review time.

CC: @nickvergessen @MorrisJobke @DeepDiver1975

**Comment by rullzer** [comment]
Ok cleaned up a bit more so we return a little less cluttered stuff.

**Comment by rullzer** [comment]
OCI failure seems unrelated.

**Comment by nickvergessen** [comment]
Didn't test, but looks good.

**Comment by PVince81** [comment]
Works :+1: 

I get the JS code will need to be adjusted later.

**Comment by nickvergessen** [comment]
PyOC patch https://github.com/owncloud/pyocclient/pull/122

**Comment by rullzer** [comment]
> I get the JS code will need to be adjusted later.

@PVince81 Already done in #19514
But I did not want to have 1 huge PR.

**Comment by PVince81** [comment]
Ah yes, the JS code update is here https://github.com/owncloud/core/pull/19514

**Comment by carlaschroder** [comment]
@rullzer Is this only for 8.2?

**Comment by nickvergessen** [comment]
yes

**Comment by carlaschroder** [comment]
thanks @nickvergessen

**Pull request merged**